### ISSUE TITLE
Personal Blog - a new Gatsby.js theme

### ISSUE THREAD

**Issue by greglobinski** [greglobinski/gatsby-starter-personal-blog#89]
Maybe someone of you already know. Last week I released my first Gatsby.js [theme](https://github.com/greglobinski/gatsby-themes/blob/master/packages/gatsby-theme-elevator-pitch/README.md)

Themes is a great new, still experimental, feature of Gatsby.js. I decided to transform to themes all of my starters, just now. 

So, it's time for **Personal Blog**, although in this  case the transformation will not be limited to implementing theme functionality. I decided to rewrite the whole codebase. 

These are goals of the transformation:
- upgrade to Gatsby v.2
- for CSS, `emotion` instead of `jss`
- for buttons and form elements, custom components instead of `material-ui`
- for dropdown menus: `@reach/menu-button`
- for form validation: `formik`
- for UI state management: React `Context` instead of `redux`

- as a cherry on top: we will rewrite  `Class` components to functional ones with `hooks`

I will start this weekend. If there is anybody who would like to help, write a comment below. Especially, I'm looking for someone who will help me rewrite styles from `jss` to `emotion`.

**Comment by maxscience** [comment]
Cool! What’s your estimate for completing the rewrite?
Wishlist:

- multiple tags
- netlify-cms (gatsby plugin)
- discus comments (or something else, just not facebook)

**Comment by hendra-go** [comment]
@greglobinski Great news Greg! thanks for your work.

After reading this, i told my self that my hunch is right that you are going to replace jss with something else, i think it is a good choice that you chose emotion, because AFAIK gatsby works nice with emotion.

But before i, we or anybody decide to contribute to rewrite jss to emotion, i think at first you need to work on replacement from redux to react context, imho.

Once again, thanks greg, it is great to know that you will continue to support this great gatsby starter.

**Comment by hendra-go** [comment]
damn, i just realize what you mean about Theme is Gatsby.js Theme feature [https://www.gatsbyjs.org/blog/2018-11-11-introducing-gatsby-themes/](https://www.gatsbyjs.org/blog/2018-11-11-introducing-gatsby-themes/), to be honest, i find out about this just now.

Haha i confuse between gatsby starter and gatsby theme.

Hmm... i need to check this out this weekend

**Comment by greglobinski** [comment]
> Cool! What’s your estimate for completing the rewrite?

I can't say, I personally can work on this project only on weekends, so for sure it will not be one weekend.  :)

> multiple tags, netlify-cms (gatsby plugin) ,discus comments (or something else, just not facebook)

all these can be done, but it's  not in the scope of this iteration

> But before i, we or anybody decide to contribute to rewrite jss to emotion, i think at first you need to work on replacement from redux to react context, imho.

I will not try to replace `redux` with `contex` trying to maintain existing code working. I will start from zero level building the site step by step. 

Update
Of course we will use the old code as a help, (mostly CSS styles) but I think that there will not be a code we would be able just copy & paste.

**Comment by greglobinski** [comment]
Update: 

We will use `emotion` in a **styled component** way

**Comment by maxscience** [comment]
Ok so I guess if one wants to start a new blog now, it will be easy to switch over the new version (once the theme is done) by just copying over the content directory right?

**Comment by greglobinski** [comment]
@maxscience Generally yes, but I can't promise that it will be possible without any 'moving' procedure, like for example, renaming content files.

**Comment by greglobinski** [comment]
If anybody would be interested in the progress of work, go to the issue board of my **[gatsby-themes repo](https://github.com/greglobinski/gatsby-themes/issues?utf8=%E2%9C%93&q=+label%3A%22%40+personal-blog-theme%22+)**

**Comment by greglobinski** [comment]
Current state of dev

![screenshot from 2019-02-23 15-06-57](https://user-images.githubusercontent.com/32480082/53287461-bc87ce80-377c-11e9-82fd-c904ee52c1eb.png)


You can follow the changes in the code [here](https://github.com/greglobinski/gatsby-themes/tree/master/packages/gatsby-theme-personal-blog)

**Comment by greglobinski** [comment]
current state of dev

![peek 2019-02-24 09-58](https://user-images.githubusercontent.com/32480082/53297195-f0b1cc80-381a-11e9-8e98-8f2695ad32b4.gif)

**Comment by greglobinski** [comment]
current state of dev

![peek 2019-02-24 15-30](https://user-images.githubusercontent.com/32480082/53300635-4782ca80-384a-11e9-8592-af69e2d3448f.gif)

That's all for this weekend, I will not (probably) be able to move forward till the next one. But I did a good progress till now, so stay tuned.

**Comment by greglobinski** [comment]
current state of dev


![peek 2019-02-25 08-33](https://user-images.githubusercontent.com/32480082/53321251-2a94d880-38d8-11e9-81b7-28f8f824a47b.gif)

**Comment by greglobinski** [comment]
current state of dev; slowly but steadily :)

![peek 2019-03-03 17-04](https://user-images.githubusercontent.com/32480082/53697906-6fd47100-3dd6-11e9-8002-1e74939a33a9.gif)

**Comment by greglobinski** [comment]
another step forward

![peek 2019-03-06 01-06](https://user-images.githubusercontent.com/32480082/53846161-30f81400-3fac-11e9-9b18-cbe3709d555e.gif)

**Comment by eclectic-coding** [comment]
Greg,
Love the look of what's happening with Gatsby and your starter/theme. I am curious. Would it be possible to integrate the gravatarMD5 feature from hero-blog? This is a really cool feature which gives a new level of functionality.

**Comment by greglobinski** [comment]
@polishedwp Sure, maybe not in the first iteration, but later why not :)

**Comment by greglobinski** [comment]
current state of development

![Peek 2019-03-11 11-57](https://user-images.githubusercontent.com/32480082/54119198-13380e00-43f5-11e9-9b44-1c2fb959c74c.gif)

although it was not planned, the theme will work with `*.mdx` files as a source of content.   https://twitter.com/greglobinski/status/1104710231605628928

**Comment by alexpov** [comment]
@greglobinski looking forward to the new theme, checking new commits every weekend :)

question/request: the original blog theme misses tags per post and a screen to filter by tags, any chance you plan to add these to your plan?

Estimates for the next release of the blog theme?

**Comment by greglobinski** [comment]
@alexpov  	 

> the original blog theme misses tags per post and a screen to filter by tags, any chance you plan to add these to your plan?

Yup, I will add tags

> Estimates for the next release of the blog theme?

Not 100% sure, but I think I need two more weekends to release the theme in its first shape.

**Comment by greglobinski** [comment]
current state of dev


![Peek 2019-03-19 10-48](https://user-images.githubusercontent.com/32480082/54596341-bbbd2200-4a34-11e9-8898-39d80ec89f6f.gif)

**Comment by wzard** [comment]
HI @greglobinski Any updates on the theme update?

**Comment by greglobinski** [comment]
@wzard Work is in progress. Not many visual changes though.

**Comment by alexpov** [comment]
Hi @greglobinski, any updates ? Looking forward for the next release ...

**Comment by h1eutran** [comment]
Hey @greglobinski, any things?

**Comment by Raf1001** [comment]
Hi all, I'm a Gatbsy newbie, been trying to run this starter straight out of the box but am getting numerous errors, is there something obvious I've missed or are there known issues with it at the moment?

Would really love to get this working as its looks fantastic!

The initial errors are these...

`
info sharp Using cached /Users/xxx/.npm/_libvips/libvips-8.7.0-darwin-x64.tar.gz
prebuild-install WARN install No prebuilt binaries found (target=12.1.0 runtime=node arch=x64 libc= platform=darwin)
  TOUCH Release/obj.target/libvips-cpp.stamp
  CXX(target) Release/obj.target/sharp/src/common.o
In file included from ../src/common.cc:27:
../src/common.h:78:20: error: no member named 'Handle' in namespace 'v8'
  bool HasAttr(v8::Handle<v8::Object> obj, std::string attr);
               ~~~~^
../src/common.h:78:37: error: expected '(' for function-style cast or type construction
  bool HasAttr(v8::Handle<v8::Object> obj, std::string attr);
                          ~~~~~~~~~~^
../src/common.h:78:39: error: use of undeclared identifier 'obj'
  bool HasAttr(v8::Handle<v8::Object> obj, std::string attr);`

Appreciate any help!

**Comment by eddiebeazer** [comment]
Try clearing public, .cache folder and using node 11

**Comment by polishedwp** [comment]
I was going to post that there are issues with `sharp` and `Node v.12` so looks like you are on the right track. 

Also, has this starter been upgraded to `Gatsby v.2` yet?

**Comment by eddiebeazer** [comment]
The starter is still on v1

**Comment by Raf1001** [comment]
Hi, thanks for the replies guys!

Yes I did already try clearing the cache but I’ll try the Node 12 and see if that works.

Regarding the V2, I thought I saw a check in last year which upgraded it to V2, could have been another branch though.

Thanks again!

**Comment by Raf1001** [comment]
Tried installing Node 11 and then got error stating Sharp model was missing so I installed it and then getting error telling me to remove the module...

Might be best to wait for Greg's V2 update.

`Something went wrong installing the "sharp" module

dlopen(/Users/xxx/Desktop/life-blog2/node_modules/sharp/build/Release/sharp.node, 1): Library not loaded: @rpath/libvips.42.dylib
  Referenced from: /Users/xxx/Desktop/life-blog2/node_modules/sharp/vendor/lib/libvips-cpp.42.dylib
  Reason: Incompatible library version: libvips-cpp.dylib requires version 52.0.0 or later, but libvips.42.dylib provides version 51.0.0

- Remove the "node_modules/sharp" directory, run "npm install" and look for errors
- Consult the installation documentation at https://sharp.pixelplumbing.com/en/stable/install/
- Search for this error at https://github.com/lovell/sharp/issues
`

**Comment by greglobinski** [comment]
@Raf1001 I'm sorry, but this thread is reserved to inform others about progress in `theme` development, so I had to remove your comments, but feel free to open a separate issue.

**Comment by greglobinski** [comment]
Hey guys, sorry for the delay but be sure that I did not abandon the project.

**Comment by h1eutran** [comment]
@greglobinski It been so long seen your last commit!

**Comment by sankhaKarunasekara** [comment]
How is it going @greglobinski. awesome theme, hope to see the new version soon. good luck. :)

**Comment by musfiqur-bs23** [comment]
HI @greglobinski , Is there any update for the new version of the theme ?
